pkg/spec/deployment_fix.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2017 The Kedge Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spec
+
+import (
+	"github.com/pkg/errors"
+)
+
+func (deployment *DeploymentSpecMod) Fix() error {
```

Comment:
code in this method will repeat for each controller, so may be create a method on `ControllerFields` and call `deployment.ControllerFields.Fix()` from here. So only this line will repeat in each controller.

Reply:
Yep, makes sense, will do this when I'll add the jobs controller.

Reply:
can you add a TODO here in code that we are gonna do this one, just so that we don't forget about it!

Reply:
done, added